gst/nnstreamer/include/nnstreamer_plugin_api_filter.h

```diff
@@ -548,6 +548,35 @@ extern accl_hw parse_accl_hw_fill (parse_accl_args accl_args);
  */
 #define parse_accl_hw(...) parse_accl_hw_fill((parse_accl_args){__VA_ARGS__})
 
+/* extern functions for shared model representation table */
+/**
+ * @brief Find the shared model representation at the table.
+ * @param[in] key The key to find the matched value of hash table.
+ * @param[out] value The matched value with inserted key. If there is no key in the hash table, it will be NULL.
+ * @return TRUE if the lookup is done. FALSE if failed to find it.
+ */
+extern int
+nnstreamer_filter_shared_table_lookup (const char *key, void **value);
+
+/* extern functions for shared model representation table */
+/**
+ * @brief Insert the new shared model representation at the table.
+ * @param[in] key The new key to store at hash table.
+ * @param[out] value The new value to store at hash table.
+ * @return TRUE if the new key and value are inserted. FALSE if failed to insert it.
+ */
+extern int
+nnstreamer_filter_shared_table_insert (char *key, void *value);
+
+/* extern functions for shared model representation table */
+/**
+ * @brief remove the shared model representation at the table.
+ * @param[in] key The key to remove the matched value of hash table.
+ * @return TRUE if the new key and value are removed. FALSE if failed to remove it.
+ */
+extern int
+nnstreamer_filter_shared_table_remove (const char *key);
```

Comment:
As I've mentioned above, remove (key, value) might be more useful than this. And realization w/ mutex becomes more robust with this semantics.

Reply:
If you feel ready, @kimjh12 you may jump in (need to talk w/ @helloahn for intentions and details of this PR and its expected following commits)

Reply:
The code has been updated to use APIs rather than to allow subplugins to access the referred list directly.
There is some update is required (mutex, hash key, reload model).
I'll share the detail with @kimjh12 before I leave and try to follow this.